File: server/router/business/cmsDiscuss.go
package business

import (
	"go-cms/api/v1"
	"go-cms/middleware"
	"github.com/gin-gonic/gin"
)

type CmsDiscussRouter struct {
}

// InitCmsDiscussRouter 初始化 CmsDiscuss 路由信息
func (s *CmsDiscussRouter) InitCmsDiscussRouter(Router *gin.RouterGroup) {
	router := Router.Group("cmsDiscuss").Use(middleware.OperationRecord())
	routerNoRecord := Router.Group("cmsDiscuss")
	var apiV1 = v1.ApiGroupApp.BusinessApiGroup.CmsDiscussApi
	{
		router.POST("createCmsDiscuss", apiV1.CreateCmsDiscuss)   // 新建CmsDiscuss
		router.POST("deleteCmsDiscuss", apiV1.DeleteCmsDiscuss) // 删除CmsDiscuss
		router.POST("deleteCmsDiscussByIds", apiV1.DeleteCmsDiscussByIds) // 批量删除CmsDiscuss
		router.POST("updateCmsDiscuss", apiV1.UpdateCmsDiscuss)    // 更新CmsDiscuss
	    router.POST("quickEdit", apiV1.QuickEdit)  // 快速编辑	
	}
	{
		routerNoRecord.GET("findCmsDiscuss", apiV1.FindCmsDiscuss)        // 根据ID获取CmsDiscuss
		routerNoRecord.GET("getCmsDiscussList", apiV1.GetCmsDiscussList)  // 获取CmsDiscuss列表
        routerNoRecord.GET("excelList", apiV1.ExcelList)  // 分页导出excel CmsDiscuss列表
	}
}

File: web/src/api/cmsDiscuss.ts
import { http,resData } from "@/utils/http";
 
// @Tags CmsDiscuss
// @Summary 创建CmsDiscuss
export const createCmsDiscuss = (data?: object) => {
   return http.request<resData>("post", "/cmsDiscuss/createCmsDiscuss", { data });  
}

// @Tags CmsDiscuss
// @Summary 删除CmsDiscuss
export const deleteCmsDiscuss = (data?: object) => {
   return http.request<resData>("post", "/cmsDiscuss/deleteCmsDiscuss", { data });  
}
 

// @Tags CmsDiscuss
// @Summary 删除ByIdsCmsDiscuss
export const deleteCmsDiscussByIds = (data?: object) => {
   return http.request<resData>("post", "/cmsDiscuss/deleteCmsDiscussByIds", { data });  
}
 

// @Tags CmsDiscuss
// @Summary 更新CmsDiscuss
export const updateCmsDiscuss = (data?: object) => {
   return http.request<resData>("post", "/cmsDiscuss/updateCmsDiscuss", { data });  
}

// @Tags CmsDiscuss
// @Summary 快速编辑
export const quickEdit = (data?: object) => {
   return http.request<resData>("post", "/cmsDiscuss/quickEdit", { data });  
}
 

// @Tags CmsDiscuss
// @Summary 用id查询CmsDiscuss
export const findCmsDiscuss = (params?: object) => {
   return http.request<resData>("get", "/cmsDiscuss/findCmsDiscuss", { params });  
}
 

// @Tags CmsDiscuss
// @Summary 分页获取CmsDiscuss列表
export const getCmsDiscussList = (params?: object) => {
   return http.request<resData>("get", "/cmsDiscuss/getCmsDiscussList", { params });  
}
 

// @Tags CmsDiscuss
// @Summary 导出excelCmsDiscuss列表
export const excelList = (params?: object) => {
   return http.request<resData>("get", "/cmsDiscuss/excelList", { params });  
}
